homework/content.js

```diff
@@ -1,37 +1,52 @@
 'use strict';
 
 (function() {
-  var _wrapper = document.querySelector('#note-content-wrapper');
+  
+  var DrawNoteManager = function(){
+  	this._wrapper = document.querySelector('#note-content-wrapper');
+  };
 
-  function start() {
-    window.addEventListener('note-open', function(event) {
-      var note = event.detail;
-      resetWrapper();
-      drawNote(note);
-    });
-  }
+  DrawNoteManager.prototype = {
+  	start() {
+	  window.addEventListener('note-open', this);
+	},
+
+	resetWrapper() {
+      this._wrapper.innerHTML = '';
+	},
 
-  function resetWrapper() {
-    _wrapper.innerHTML = '';
-  }
+	noteOpen(event){
+      var note = event.detail;
+      var promise = Promise.resolve();
+      promise.then(this.resetWrapper()).then(this.drawNote(note));
+	},
 
-  function drawNote(note) {
-    var title = note.title;
-    var h = document.createElement('h2');
-    h.textContent = title;
-    var passages = note.passages;
-    var buff = document.createDocumentFragment();
-    passages.forEach(function(passage) {
-      var p = document.createElement('p');
-      p.classList.add('note-passage');
-      p.textContent = passage;
-      buff.appendChild(p);
-    });
-    _wrapper.appendChild(h);
-    _wrapper.appendChild(buff);
-  }
+	drawNote(note) {
+      var title = note.title;
+      var h = document.createElement('h2');
+      h.textContent = title;
+      var passages = note.passages;
+	  var buff = document.createDocumentFragment();
+	  passages.forEach(function(passage) {
+	    var p = document.createElement('p');
+	    p.classList.add('note-passage');
+	    p.textContent = passage;
+	    buff.appendChild(p);
+	  });
+	  this._wrapper.appendChild(h);
+	  this._wrapper.appendChild(buff);
+	},
 
+	handleEvent(event) {
+      switch (event.type) {
+        case 'note-open':
+          this.noteOpen(event);
+          break;
+      }
+    }
+  };
   document.addEventListener('DOMContentLoaded', function(event) {
```

Comment:
It's better to collect all managers in one startup method, rather than scatter them about two files.